src/irr/asset/CMeshManipulator.cpp

```diff
@@ -1551,6 +1551,343 @@ core::smart_refctd_ptr<ICPUBuffer> IMeshManipulator::idxBufferFromTrianglesFanTo
+
+        p0 = minVtxPos[bestPairIdx];
+        p1 = maxVtxPos[bestPairIdx];
+
+        if (core::distancesquared(p0, p1).x < eps)
+        {
+            // return AABB
+            return constructOBB(core::matrix3x4SIMD(), AABBlen / 2.0f, core::vectorSIMDf());
+        }
+
+        //TODO: check it!!
+        auto pointToLineDistanceSquared = [&](const core::vectorSIMDf p0, const core::vectorSIMDf dir, const core::vectorSIMDf v)
+        {
+            _NBL_DEBUG_BREAK_IF(core::length(dir).x < eps);
+
+            const core::vectorSIMDf u = v - p0;
+            return core::lengthsquared(u - (dir * core::dot(u, dir).x)).x;
+        };
+
+        e0 = core::normalize(p0 - p1);
+
+        // Find a third vertex furthest away from line given by p0, e0
+        maxDistance = pointToLineDistanceSquared(p0, e0, selVtxPos[0]);
+        p2 = selVtxPos[0];
+        for (uint32_t i = 1u; i < np; i++)
+        {
+            float distance = pointToLineDistanceSquared(p0, e0, selVtxPos[i]);
+
+            if (distance > maxDistance)
+            {
+                p2 = selVtxPos[i];
+                maxDistance = distance;
+            }
+        }
+
+        //TODO: handle this case
+        if (maxDistance < eps)
+        {
+            _NBL_DEBUG_BREAK_IF(true);
+        }
+
+        // calculate edges and normal of the base triangle
+        e1 = core::normalize(p1 - p2);
+        e2 = core::normalize(p2 - p0);
+        n = core::normalize(core::cross(e1, e0));
+    }
+
+    //find remaining vertices of ditetrahedron
+    {
+        //find vertices that are furthest from the plane defined by p0, p1 and p2 (base triangle) on the both positive and negative half space
+        //n dot x = d
+        const float d = core::dot(p0, n).x;
+
+        float minDistance;
+        float maxDistance;
+        minDistance = maxDistance = core::dot(mb->getPosition(0u), n).x - d;
+        t0 = t1 = mb->getPosition(0u);
+
+        for (size_t i = 0; i < mb->calcVertexCount(); i++)
+        {
+            float distance = core::dot(mb->getPosition(i), n).x - d;
+
+            if (distance < minDistance)
+            {
+                minDistance = distance;
+                t0 = mb->getPosition(i);
+            }
+            if (distance > maxDistance)
+            {
+                maxDistance = distance;
+                t1 = mb->getPosition(i);
+            }
+        }
+    }
+
+    auto findExternalPointProj = [&](const core::vectorSIMDf& dir, float& minPoint, float& maxPoint)
+    {
+        minPoint = maxPoint = core::dot(selVtxPos[0], dir).x;
+
+        for (size_t i = 1u; i < np; i++)
+        {
+            float proj = core::dot(selVtxPos[i], dir).x;
+            
+            if (proj > maxPoint)
+                maxPoint = proj;
+
+            if (proj < minPoint)
+                minPoint = proj;
+        }
+    };
+
+    auto findImprovedAxesFromTriangle = [&](const core::vectorSIMDf& v0, const core::vectorSIMDf& v1, const core::vectorSIMDf& v2, 
```

Comment:
this should only take 3 vertices (unnormalized)

you can do the rest of the legwork here